### ISSUE TITLE
TLS1.3 Support for Websockets?

### ISSUE THREAD

**Issue by tomtastic** [owntracks/ios#633]
I run an MQTT server with secure websockets, Owntracks iOS can no longer connect if I restrict the minimum TLS version to 1.3.

Is it possible to add support?

**Comment by ckrey** [comment]
The version currently in the app store uses facebook's SocketRocket library for websockets.
Unfortunately, the library is no longer maintained. This might be the reason for what you see with regards to minimum TLS version 1.3

A completely new approach is in the making which uses Apple's native websocket implementation.
This will be available with a minimum iOS version 13.x. We will ship this version later this year.

**Comment by tomtastic** [comment]
Thanks 🙏, I'll look forward to that.

**Comment by ckrey** [comment]
@tomtastic I just tested the current app store version of OwnTracks against mosquitto 1.6.8 running websockets on top of tlsv1.3 and I do see no problem. 

```
...
listener 8088 0.0.0.0
protocol websockets
tls_version 1.3
cafile /etc/letsencrypt/...
```

How and in which environment do you `restrict the minimum TLS version to 1.3.`

**Comment by tomtastic** [comment]
For some clarification, my TLS is terminated by [Traefik 2.x](https://traefik.io) which forwards the non-TLS Websocket traffic on to [Eclipse Mosquitto 1.6.x](https://mosquitto.org/).

TLS minimum version is set within Traefik thusly :
```
tls:
  options:
    mintls13:
      minVersion: VersionTLS13
``` 
and the Mosquitto container picks that up with :
```
  mosquitto:
    image: eclipse-mosquitto:1.6
    expose:
      - 8083
    labels:
      traefik.enable: true
      traefik.http.routers.mosquitto.entryPoints: "https"
      traefik.http.routers.mosquitto.tls: true
      traefik.http.routers.mosquitto.tls.options: "mintls13@file"
```

**Comment by ckrey** [comment]
Testing OwnTracks 13.1.7:

- wss -> mosquitto tlsv1.3: works
- wss -> apache2 tlsv1.3: works
The proxy setup here is similar to your `Traefik` configuration. 

- mqtts -> mosquitto tlsv1.3: DOES NOT WORK
- mqtts -> mosquitto tlsv1.2: works

Testing OwnTracks 13.3.x (not yet available in App Store):
- all combinations work

**Comment by ckrey** [comment]
available in versions >= 13.3.6
